Source/missiles.cpp

```diff
@@ -3004,6 +3004,485 @@ void MI_Fireball(int i)
+				    && monster[missile[i]._misource].MType->mtype <= MT_MAEL) {
+					AddMissile(
+					    mx,
+					    my,
+					    missile[i]._misx,
+					    missile[i]._misy,
+					    i,
+					    MIS_LIGHTNING2,
+					    missile[i]._micaster,
+					    missile[i]._misource,
+					    dam,
+					    missile[i]._mispllvl);
+				} else {
+					AddMissile(
+					    mx,
+					    my,
+					    missile[i]._misx,
+					    missile[i]._misy,
+					    i,
+					    MIS_LIGHTNING,
+					    missile[i]._micaster,
+					    missile[i]._misource,
+					    dam,
+					    missile[i]._mispllvl);
+				}
+			} else {
+				AddMissile(
+				    mx,
+				    my,
+				    missile[i]._misx,
+				    missile[i]._misy,
+				    i,
+				    MIS_LIGHTNING,
+				    missile[i]._micaster,
+				    missile[i]._misource,
+				    dam,
+				    missile[i]._mispllvl);
+			}
+			missile[i]._miVar1 = missile[i]._mix;
+			missile[i]._miVar2 = missile[i]._miy;
+		}
+	}
+
+	if (missile[i]._mirange == 0 || mx <= 0 || my <= 0 || mx >= MAXDUNX || my > MAXDUNY) {
+		missile[i]._miDelFlag = TRUE;
+	}
+}
+
+void mi_flashfr(int i)
+{
+	int src;
+
+	src = missile[i]._misource;
+	if (!missile[i]._micaster && src != -1) {
+		missile[i]._mix = plr[src].WorldX;
+		missile[i]._miy = plr[src].WorldY;
+		missile[i]._mitxoff = plr[src]._pxoff << 16;
+		missile[i]._mityoff = plr[src]._pyoff << 16;
+	}
+	missile[i]._mirange--;
+	if (!missile[i]._mirange) {
+		missile[i]._miDelFlag = TRUE;
+		if (!missile[i]._micaster) {
+			src = missile[i]._misource;
+			if (src != -1)
+				plr[src]._pBaseToBlk -= 50;
+		}
+	}
+	PutMissile(i);
+}
+
+void mi_flashbk(int i)
+{
+	if (!missile[i]._micaster) {
+		if (missile[i]._misource != -1) {
+			missile[i]._mix = plr[missile[i]._misource]._px;
+			missile[i]._miy = plr[missile[i]._misource]._py;
+		}
+	}
+	missile[i]._mirange--;
+	if (!missile[i]._mirange)
+		missile[i]._miDelFlag = TRUE;
+	PutMissile(i);
+}
+
+void mi_reflect(int i)
+{
+	int src;
+
+	src = missile[i]._misource;
+	missile[i]._mitxoff = plr[src]._pxoff << 16;
+	missile[i]._mityoff = plr[src]._pyoff << 16;
+	if (plr[src]._pmode == PM_WALK3) {
+		missile[i]._misx = plr[src]._px + 2;
+		missile[i]._misy = plr[src]._py - 1;
+	} else {
+		missile[i]._misx = plr[src].WorldX + 2;
+		missile[i]._misy = plr[src].WorldY - 1;
+	}
+	GetMissilePos(i);
+	if (plr[src]._pmode == PM_WALK3) {
+		if (plr[src]._pdir == DIR_W)
+			++missile[i]._mix;
+		else
+			++missile[i]._miy;
+	}
+	if (src != myplr && currlevel != plr[src].plrlevel)
+		missile[i]._miDelFlag = TRUE;
+	if ((WORD)plr[src].wReflection <= 0) {
+		missile[i]._miDelFlag = TRUE;
+		NetSendCmd(TRUE, 0x63u); //TODO: apply enums
+	}
+	PutMissile(i);
+}
+
+void mi_fire_ring(int i)
+{
+}
+
+void mi_light_ring(int i)
+{
+}
+
+void mi_search(int i)
+{
+	missile[i]._mirange--;
+	if (!missile[i]._mirange) {
+		missile[i]._miDelFlag = TRUE;
+		PlaySfxLoc(87, plr[missile[i]._miVar1].WorldX, plr[missile[i]._miVar1].WorldY);

```

Comment:
```suggestion
		PlaySfxLoc(IS_CAST6, plr[missile[i]._miVar1].WorldX, plr[missile[i]._miVar1].WorldY);
```

Reply:
your suggestion was wrong, it's IS_CAST7 :D